### DISCUSSION TITLE
Passing subscribers to Typeorm

### DISCUSSION THREAD

**Discussion by khanh-to-niteco** [medusajs/medusa#6811] [Feature Requests]
Hi Medusa team,

In our project, we have a need to register custom subscribers to Typeorm to extend Medusa with some custom logic

![image](https://github.com/medusajs/medusa/assets/71063731/2bd5f338-f5a4-489e-b07c-2f576ddd31e4)

But we cannot pass this parameter to Typeorm DataSourceOptions without modifying Medusa.

![image](https://github.com/medusajs/medusa/assets/71063731/fdc103e3-da1a-4ab0-b303-1d317e573ebe)

Could you please help to check and support this?

Thanks

**Comment by olivermrbl** [comment]
Hi @khanh-to-niteco, could I get you to share the use case? Ideally, we don't extend the core entities with custom subscribers, so I would like to see if there are ways around this.

**Reply by khanh-to-niteco** [reply]
Hi @olivermrbl ,

We need to add some more fields to orders. But we are unable to pass these fields from cart. If you look at the `orderService.createFromCart` method:

![image](https://github.com/medusajs/medusa/assets/71063731/709c3054-8688-4f66-b61a-568eb80ad5ad)

Only some fields are copied from cart to order. One of them is the `metadata` field, so our quick fix was to use the metadata field of cart to carry these additional fields.

And in the event subscribers of Typeorm, we copy these values in `order.metadata` to its new fields.
